### ISSUE TITLE
ERROR: Failed building wheel for dependency-injector

### ISSUE THREAD

**Issue by leonyap27** [aiverify-foundation/moonshot#311]
![image](https://github.com/user-attachments/assets/e4b763ef-97d4-4042-8d4e-2f010da8319f)
Attempt to install the requested package, however it still hit the same error message when I re-run the following command: pip install "aiverify-moonshot[all]".

Kindly assist,
Leon

**Comment by imda-normanchia** [comment]
Hi Leon,

According to the log person that you have send, you might have an older version of Microsoft Visual C++. Are you able to check the version of your C+= build tools?

**Comment by leonyap27** [comment]
Thank! I managed to get it running. 
1) It freezes on the Run Screen when I attempt to test a model. 
![image](https://github.com/user-attachments/assets/cb04eb15-37b6-4b07-8c45-10019d9b8d20)

2) Would like to verify for OpenAI GPT4 the API in the following image below? 
![Uploading image.png…]()

**Comment by leonyap27** [comment]
You may refer to the following log below
![image](https://github.com/user-attachments/assets/99f418b6-24bf-4374-af3e-bf0d99442cee)

Kindly assist,
Leon

**Comment by imda-normanchia** [comment]
in order to generate a result for `mlc-ai-safety` , you will need to have an API token for `together-llama-guard-7b-assistant` . maybe you can try  excluding the MLC Safety cookbook from your benchmark run

**Comment by imda-normanchia** [comment]
also it seems that you might be missing the openai. maybe certain dependencies weren't installed fully when you faced the issue with tensorflow. maybe you can try re installing the requirements for moonshot-data

**Comment by imda-normanchia** [comment]
hello @leonyap27  ,

Are you still facing any issue with running moonshot?

**Comment by leonyap27** [comment]
> hello @leonyap27 ,
> 
> Are you still facing any issue with running moonshot?
Yes, I try re-install on another workstation I encounter the issue below (see image 1), seem like it require tensorflow-io-gcs-filesystem 0.37.1 but (refer to image 2) in anaconda.org only have 3.10 available. 

image 1
![image](https://github.com/user-attachments/assets/0a4a8260-cbea-4f94-92dc-56762e8f9685)
image 2
![image](https://github.com/user-attachments/assets/c2cc5933-1f74-44e2-9842-aafc1614f2f9)

**Comment by imda-normanchia** [comment]
may i know what version of python and pip3 you are running on?

**Comment by leonyap27** [comment]
> may i know what version of python and pip3 you are running on?
@imda-normanchia , I manage to install 3.10 refer to https://github.com/aiverify-foundation/moonshot/issues/312.  

However I encounter runtime issue, I believe maybe I not familiar with the configuration. 
If possible can provide the high level information where I can get relevant API key for different LLM. Thanks. 
![image](https://github.com/user-attachments/assets/888f03e1-c110-4c83-a374-3d151ffcf4f9)

**Comment by imda-normanchia** [comment]
if you are testing open ai model, you would need to put in your OpenAI API Key in the token field. 
you may refer to this article on how to find your api key. 

https://help.openai.com/en/articles/4936850-where-do-i-find-my-openai-api-key

**Comment by imda-normanchia** [comment]
hello, may i know if you are still facing any issue? i would be closing this issue if you are no longer facing any issue.
